### DISCUSSION TITLE
Question about Foundation website and smart contract interaction

### DISCUSSION THREAD

**Discussion by tomHsieh85** [f8n/fnd-protocol#27] [Q&A] [answered]
Hello!

Thanks for sharing the foundation contract. I am curious which smart contract does foundation web app interact with to render, for example the https://foundation.app/collections page? Are those data stored on-chain or off-chain? I understand that individual NFT data is stored on-chain but I am curious about if the search function is interact with on-chain data or off-chain data?

**Comment by HardlyDifficult** [comment, marked as answer]
Hey!
Our search features are done off-chain ATM by caching the on-chain data. We do have a subgraph that may help with a few of these, we'll have that deployed to the decentralized subgraph soon where you could leverage that as well.
